Source/Core/Parser.cs

```diff
@@ -511,7 +512,7 @@ public override int ComputeHashCode()
 				Spec(pre, mods, post);
 			}
 			ImplBody(out locals, out stmtList);
-			impl = new Implementation(x, x.val, typeParams,
+			impl = new Implementation(x, x.val, new List<TypeVariable>(typeParams.Select(x => new TypeVariable(x.tok, x.Name))),
```

Comment:
If you use `ConvertAll` instead of `Select`, you don't need `using System.Linq;`.

```suggestion
			impl = new Implementation(x, x.val, typeParams.ConvertAll(x => new TypeVariable(x.tok, x.Name)),
```

Reply:
Done.